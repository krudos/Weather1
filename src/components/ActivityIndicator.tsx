import React from 'react';
import {View} from 'react-native';
import {ActivityIndicator as ActivityIndicatorPaper} from 'react-native-paper';

interface ActivityIndicatorProps {
  isLoading: boolean;
}
const ActivityIndicator = (props: ActivityIndicatorProps) => {
  if (!props.isLoading) {
    return null;
  }
  return (
    <View
      style={{
        justifyContent: 'center',
        alignItems: 'center',
        flex: 1,
        flexGrow: 1,
      }}>
      <ActivityIndicatorPaper />
    </View>
  );
};
export {ActivityIndicator};
