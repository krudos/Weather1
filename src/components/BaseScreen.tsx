import React from 'react';
import {PropsWithChildren} from 'react';
import {ScrollView, StyleSheet} from 'react-native';
import {Edges, SafeAreaView} from 'react-native-safe-area-context';

type BaseScreenProps = PropsWithChildren<{edges?: Edges | undefined}>;

const BaseScreen = ({children, edges}: BaseScreenProps) => {
  return (
    <SafeAreaView style={styles.scrollView} edges={edges}>
      <ScrollView>{children}</ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  scrollView: {flex: 1, paddingLeft: 10, paddingRight: 10},
});

export {BaseScreen};
