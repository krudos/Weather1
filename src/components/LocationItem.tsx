import React from 'react';

import {TouchableOpacity, View} from 'react-native';
import {Location} from '../types/Location';
import {Text} from 'react-native-paper';
import CountryFlag from 'react-native-country-flag';
import {countryToAlpha2} from 'country-to-iso';

interface LocationItemProps {
  location: Location;
  onPress: () => void;
}

const LocationItem = ({location, onPress}: LocationItemProps) => {
  const country = countryToAlpha2(location.country);
  return (
    <TouchableOpacity
      onPress={onPress}
      style={{
        flexDirection: 'row',
        marginTop: 10,
        marginBottom: 10,
      }}>
      <View style={{flex: 1}}>
        <Text>{location.name},</Text>
        <Text>{location.country}</Text>
      </View>
      <View style={{alignItems: 'center', justifyContent: 'center'}}>
        {country && <CountryFlag isoCode={country} size={25} />}
      </View>
    </TouchableOpacity>
  );
};

export {LocationItem};
