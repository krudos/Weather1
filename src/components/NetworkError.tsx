import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Button, Text} from 'react-native-paper';

interface NetworkErrorProps {
  refetch: () => void;
  isError: boolean;
  isLoading: boolean;
}

const NetworkError = (props: NetworkErrorProps) => {
  if (props.isLoading) {
    return null;
  }

  if (!props.isError) {
    return null;
  }
  return (
    <View style={styles.content}>
      <Text>Network error</Text>
      <Button onPress={() => props.refetch()}> Retry</Button>
    </View>
  );
};

const styles = StyleSheet.create({
  content: {flex: 1, justifyContent: 'center', alignItems: 'center'},
});

export {NetworkError};
