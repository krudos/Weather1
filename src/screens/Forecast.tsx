import React from 'react';
import {Text, View} from 'react-native';

const ForecastScreen = ({route}) => {
  return (
    <View>
      <Text>{route.params.location.url}</Text>
    </View>
  );
};

export {ForecastScreen};
