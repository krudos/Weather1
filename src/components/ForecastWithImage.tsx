import React from 'react';
import {Text, View} from 'react-native';
import {Image} from 'react-native';
import {ForecastWithImageProps} from '../screens/Forecast';

export const ForecastWithImage = ({
  icon,
  condition,
  temp_c,
}: ForecastWithImageProps) => {
  return (
    <View style={{flexDirection: 'row'}}>
      <Image
        style={{width: 64, height: 64}}
        source={{
          uri: `https:${icon}`,
        }}
      />
      <View style={{justifyContent: 'center'}}>
        <Text>{condition}</Text>
        <Text>{temp_c}Â°</Text>
      </View>
    </View>
  );
};
