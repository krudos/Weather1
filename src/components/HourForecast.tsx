import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {ForecastWithImage} from './ForecastWithImage';
import {Hour} from '../types/Forecast';

const HourForecast = ({item}: {item: Hour}) => {
  return (
    <View style={styles.contentView}>
      <Text style={styles.fullView}>{item.time.substring(11)}</Text>
      <ForecastWithImage
        icon={item.condition.icon}
        condition={item.condition.text}
        temp_c={item.temp_c}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  contentView: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  fullView: {flex: 1},
});

export {HourForecast};
