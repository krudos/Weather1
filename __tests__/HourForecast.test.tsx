import React from 'react';
import {render} from '@testing-library/react-native';
import {HourForecast} from '../src/components/HourForecast';
import {Hour} from '../src/types/Forecast';

describe('HourForecast', () => {
  it('renders correctly', async () => {
    const item: Hour = {
      time_epoch: 1705640400,
      time: '2024-01-18 23:00',
      temp_c: 14.7,
      temp_f: 58.4,
      is_day: 0,
      condition: {
        text: 'Clear',
        icon: '//cdn.weatherapi.com/weather/64x64/night/113.png',
        code: 1000,
      },
      wind_mph: 3.4,
      wind_kph: 5.4,
      wind_degree: 338,
      wind_dir: 'NNW',
      pressure_mb: 1018,
      pressure_in: 30.06,
      precip_mm: 0,
      precip_in: 0,
      snow_cm: 0,
      humidity: 80,
      cloud: 15,
      feelslike_c: 14.9,
      feelslike_f: 58.8,
      windchill_c: 14.9,
      windchill_f: 58.8,
      heatindex_c: 14.7,
      heatindex_f: 58.4,
      dewpoint_c: 11.2,
      dewpoint_f: 52.2,
      will_it_rain: 0,
      chance_of_rain: 0,
      will_it_snow: 0,
      chance_of_snow: 0,
      vis_km: 10,
      vis_miles: 6,
      gust_mph: 7,
      gust_kph: 11.3,
      uv: 1,
      short_rad: 0,
      diff_rad: 0,
    };

    const tree = render(<HourForecast item={item} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
