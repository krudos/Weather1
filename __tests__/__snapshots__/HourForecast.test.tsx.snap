// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourForecast renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      {
        "flex": 1,
      }
    }
  >
    23:00
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://cdn.weatherapi.com/weather/64x64/night/113.png",
        }
      }
      style={
        {
          "height": 64,
          "width": 64,
        }
      }
    />
    <View
      style={
        {
          "justifyContent": "center",
        }
      }
    >
      <Text>
        Clear
      </Text>
      <Text>
        14.7
        Â°
      </Text>
    </View>
  </View>
</View>
`;
